src/pyuvsim/uvsim.py

```diff
@@ -784,49 +780,81 @@ def run_uvdata_uvsim(
     comm = mpi.get_comm()
     Npus = mpi.get_Npus()
 
-    input_uv = comm.bcast(input_uv, root=0)
-    beam_dict = comm.bcast(beam_dict, root=0)
-    catalog.share(root=0)
-    beam_list.share(root=0)
-
-    if not isinstance(input_uv, UVData):
-        raise TypeError("input_uv must be UVData object")
+    # ensure that we have full or flat spectral type and that we have points not
+    # healpix
+    if rank == 0:
+        if not isinstance(input_uv, UVData):
+            raise TypeError("input_uv must be UVData object")
 
-    if beam_list.beam_type != "efield":
-        raise ValueError("Beam type must be efield!")
+        if beam_list.beam_type != "efield":
+            raise ValueError("Beam type must be efield!")
 
-    if (
-        beam_list.data_normalization is not None
-        and beam_list.data_normalization != "peak"
-    ):
-        raise ValueError("UVBeams must be peak normalized.")
+        if (
+            beam_list.data_normalization is not None
+            and beam_list.data_normalization != "peak"
+        ):
+            raise ValueError("UVBeams must be peak normalized.")
+
+        exp_npols = beam_list[0].Nfeeds ** 2
+        try:
+            exp_pols = uvutils.pol.convert_feeds_to_pols(
+                feed_array=beam_list[0].feed_array,
+                include_cross_pols=True,
+                x_orientation=beam_list.x_orientation,
+            )
+        except AttributeError:
+            from pyuvdata.uvbeam.uvbeam import _convert_feeds_to_pols
 
-    exp_npols = beam_list[0].Nfeeds ** 2
-    try:
-        exp_pols = uvutils.pol.convert_feeds_to_pols(
-            feed_array=beam_list[0].feed_array,
-            include_cross_pols=True,
-            x_orientation=beam_list.x_orientation,
-        )
-    except AttributeError:
-        from pyuvdata.uvbeam.uvbeam import _convert_feeds_to_pols
+            exp_pols, _ = _convert_feeds_to_pols(
+                feed_array=beam_list[0].feed_array,
+                calc_cross_pols=bool(exp_npols > 1),
+                x_orientation=beam_list.x_orientation,
+            )
 
-        exp_pols, _ = _convert_feeds_to_pols(
-            feed_array=beam_list[0].feed_array,
-            calc_cross_pols=bool(exp_npols > 1),
-            x_orientation=beam_list.x_orientation,
-        )
+        if not (
+            (input_uv.Npols == exp_npols)
+            and (input_uv.polarization_array.tolist() == exp_pols.tolist())
+        ):
+            exp_pols_str = uvutils.polnum2str(exp_pols)
+            raise ValueError(f"input_uv must have polarizations: {exp_pols_str}")
```

Comment:
How do we feel about this error message? I feel like our errors are normally a little more verbose/informative. I think I imagined something like "Input UVData object/simulation parameters and beams polarizations do not agree. Input beams have output polarizations: {}, Simulation has expected polarizations {}"

Reply:
done.